File: 2026_GFG_POTD/JANUARY/07-01-2026/Count distinct elements in every window.js
class Solution {
    countDistinct(arr, k) {
        const res = [];
        const freq = new Map();
        
        for (let i = 0; i < arr.length; i++) {
            freq.set(arr[i], (freq.get(arr[i]) || 0) + 1);
            
            if (i >= k - 1) {
                res.push(freq.size);
                
                const left = arr[i - k + 1];
                freq.set(left, freq.get(left) - 1);
                if (freq.get(left) === 0) freq.delete(left);
            }
        }
        
        return res;
    }
}

File: 2026_GFG_POTD/JANUARY/07-01-2026/Count distinct elements in every window.java
import java.util.*;

class Solution {
    ArrayList<Integer> countDistinct(int arr[], int k) {
        ArrayList<Integer> res = new ArrayList<>();
        HashMap<Integer, Integer> freq = new HashMap<>();
        
        for (int i = 0; i < arr.length; i++) {
            freq.put(arr[i], freq.getOrDefault(arr[i], 0) + 1);
            
            if (i >= k - 1) {
                res.add(freq.size());
                
                int left = arr[i - k + 1];
                freq.put(left, freq.get(left) - 1);
                if (freq.get(left) == 0) {
                    freq.remove(left);
                }
            }
        }
        
        return res;
    }
}